### PR TITLE
feat: default displayable trace in repr and str for CallTreeNode

### PR DESCRIPTION
**Pull request by unparalleled-js**
### What I did

Provide default tree via `__repr__` and `__str__` methods.

### How I did it

Inspired from the answer with 66 updoots: https://stackoverflow.com/questions/9727673/list-directory-tree-structure-in-python

### How to verify it

print(tree)

### Checklist

- [ ] Passes all linting checks (pre-commit and CI jobs)
- [ ] New test cases have been added and are passing
- [ ] Documentation has been updated
- [ ] PR title follows [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) standard (will be automatically included in the changelog)

### CODE REVIEW

File: evm_trace/base.py [hunk 1]
```diff
@@ -58,6 +58,15 @@ class CallTreeNode(BaseModel):
     def validate_hexbytes(cls, v) -> HexBytes:
         return _convert_hexbytes(cls, v)
 
+    def __str__(self) -> str:
+        spacing = self.depth * "  "
+        title = f"{spacing} {self.call_type} call {self.address.hex()} gas={self.gas_cost}"
```
Comment (unparalleled-js) [inline evm_trace/base.py:63]: tell me any extra useful data you'd like per default node

File: tests/test_trace_frame.py [hunk 2]
```diff
@@ -33,17 +33,12 @@ def test_trace_frame_validation_passes():
...
-def test_trace_frame_validation_fails(test_value):
-    trace_frame_structure = deepcopy(TRACE_FRAME_STRUCTURE)
-    trace_frame_structure.update(test_value)
-
+))
+def test_trace_frame_validation_fails(test_data):
+    data = {**TRACE_FRAME_STRUCTURE, **test_data}
     with pytest.raises(ValidationError):
-        TraceFrame(**trace_frame_structure)
+        TraceFrame(**data)
```
Comment (NotPeopling2day) [inline tests/test_trace_frame.py:44]: Thank you!

**Comment by unparalleled-js** [comment]
default traces be like

```
Mutable call @ 0xf2df0b975c0c9efa2f8ca0491c2d1685104d2488 gas=194827
├── Mutable call @ 0xbcf7fffd8b256ec51a36782a52d0c34f6474d951 gas=168423
│   └── Mutable call @ 0x274b028b03a250ca03644e6c578d81f019ee1323 gas=160842
├── Mutable call @ 0xbcf7fffd8b256ec51a36782a52d0c34f6474d951 gas=116942
│   └── Mutable call @ 0x274b028b03a250ca03644e6c578d81f019ee1323 gas=114595
└── Mutable call @ 0xbcf7fffd8b256ec51a36782a52d0c34f6474d951 gas=93421
    ├── Mutable call @ 0x274b028b03a250ca03644e6c578d81f019ee1323 gas=91277
    ├── Mutable call @ 0x274b028b03a250ca03644e6c578d81f019ee1323 gas=46476
    └── Mutable call @ 0x274b028b03a250ca03644e6c578d81f019ee1323 gas=23491

Mutable call @ 0xf2df0b975c0c9efa2f8ca0491c2d1685104d2488 gas=228635
├── Mutable call @ 0xbcf7fffd8b256ec51a36782a52d0c34f6474d951 gas=180061
│   └── Mutable call @ 0x274b028b03a250ca03644e6c578d81f019ee1323 gas=172298
├── Mutable call @ 0xbcf7fffd8b256ec51a36782a52d0c34f6474d951 gas=145417
│   ├── Mutable call @ 0x274b028b03a250ca03644e6c578d81f019ee1323 gas=142461
│   ├── Mutable call @ 0x274b028b03a250ca03644e6c578d81f019ee1323 gas=117249
│   └── Mutable call @ 0x274b028b03a250ca03644e6c578d81f019ee1323 gas=94264
├── Mutable call @ 0xbcf7fffd8b256ec51a36782a52d0c34f6474d951 gas=73514
│   ├── Mutable call @ 0x274b028b03a250ca03644e6c578d81f019ee1323 gas=71681
│   ├── Mutable call @ 0x274b028b03a250ca03644e6c578d81f019ee1323 gas=48537
│   └── Mutable call @ 0x274b028b03a250ca03644e6c578d81f019ee1323 gas=25552
└── Static call @ 0xbcf7fffd8b256ec51a36782a52d0c34f6474d951 gas=3805
```

File: evm_trace/_display.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+if TYPE_CHECKING:
+    from evm_trace.base import CallTreeNode
+
+
+class DisplayableCallTreeNode(object):
+    _FILE_MIDDLE_PREFIX = "├──"
+    _FILE_LAST_PREFIX = "└──"
+    _PARENT_PREFIX_MIDDLE = "    "
+    _PARENT_PREFIX_LAST = "│   "
```
Comment (fubuloubu) [inline evm_trace/_display.py:11]: Completely eschewing `rich.tree`?

File: evm_trace/_display.py [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    @property
+    def depth(self) -> int:
+        return self.call.depth
+
+    @property
+    def title(self) -> str:
+        call_type = self.call.call_type.value.lower().capitalize()
+        address = self.call.address.hex()
+        cost = self.call.gas_cost
+        return f"{call_type} call @ {address} gas_cost={cost}"
```
Comment (fubuloubu) [inline evm_trace/_display.py:32]: Might tighten things up a bit to show it like this:
```suggestion
        call_type = self.call.call_type.value.upper()
        call_mnemonic = "CALL" if self.call.call_type == CallType.MUTABLE else f"{call_type}CALL"
        address = self.call.address.hex()
        cost = self.call.gas_cost
        return f"{call_mnemonic}: {address} [{cost} gas]"
```

would display like:
```
CALL: 0xf2df0b975c0c9efa2f8ca0491c2d1685104d2488 [194827 gas]
├── CALL: 0xbcf7fffd8b256ec51a36782a52d0c34f6474d951 [168423 gas]
│   └── CALL: 0x274b028b03a250ca03644e6c578d81f019ee1323 [160842 gas]
├── CALL: 0xbcf7fffd8b256ec51a36782a52d0c34f6474d951 [116942 gas]
│   └── CALL: 0x274b028b03a250ca03644e6c578d81f019ee1323 [114595 gas]
└── CALL: 0xbcf7fffd8b256ec51a36782a52d0c34f6474d951 [93421 gas]
    ├── CALL: 0x274b028b03a250ca03644e6c578d81f019ee1323 [91277 gas]
    ├── CALL: 0x274b028b03a250ca03644e6c578d81f019ee1323 [46476 gas]
    └── CALL: 0x274b028b03a250ca03644e6c578d81f019ee1323 [23491 gas]

CALL: 0xf2df0b975c0c9efa2f8ca0491c2d1685104d2488 [228635 gas]
├── CALL: 0xbcf7fffd8b256ec51a36782a52d0c34f6474d951 [180061 gas]
│   └── CALL: 0x274b028b03a250ca03644e6c578d81f019ee1323 [172298 gas]
├── CALL: 0xbcf7fffd8b256ec51a36782a52d0c34f6474d951 [145417 gas]
│   ├── CALL: 0x274b028b03a250ca03644e6c578d81f019ee1323 [142461 gas]
│   ├── CALL: 0x274b028b03a250ca03644e6c578d81f019ee1323 [117249 gas]
│   └── CALL: 0x274b028b03a250ca03644e6c578d81f019ee1323 [94264 gas]
├── CALL: 0xbcf7fffd8b256ec51a36782a52d0c34f6474d951 [73514 gas]
│   ├── CALL: 0x274b028b03a250ca03644e6c578d81f019ee1323 [71681 gas]
│   ├── CALL: 0x274b028b03a250ca03644e6c578d81f019ee1323 [48537 gas]
│   └── CALL: 0x274b028b03a250ca03644e6c578d81f019ee1323 [25552 gas]
└── STATICCALL: 0xbcf7fffd8b256ec51a36782a52d0c34f6474d951 [3805 gas]
```

File: evm_trace/_display.py (hunk 3 above)
Comment (unparalleled-js) [reply evm_trace/_display.py:11]: Only in this package! We will use it in ape, as it has lots of nice features such as  markup terminal tags. But I didn't want to take on the dependency here.

**Review by fubuloubu — APPROVED** [review]

File: setup.py [hunk 5]
```diff
@@ -56,7 +56,7 @@
         "importlib-metadata ; python_version<'3.8'",
         "pydantic>=1.9.0,<2.0",
         "hexbytes>=0.2.2,<1.0.0",
-        "eth-utils==1.10.0",
+        "eth-utils>=1.10.0",
```
Comment (fubuloubu) [inline setup.py:59]: might need to add `eth-hash[pysha3]` to `test` extras, and then make a note that downstream should also install it here

Comment (unparalleled-js) [reply setup.py:59]: Where does this get used?

**Pull request merged**